### PR TITLE
Separate _verify_single_precompiled_checksum()

### PR DESCRIPTION
**Pull request by pirapira**
This commit separates a function that checks one entry of
contracts.json.  This eases the implementation of #1257.

### What this PR does

This is a pure refactoring that separates away a subroutine.

Before this PR, ContractSourceManager didn't have a function that checks a single entry of `contracrts.json`.  There was only `verify_precompiled_checksums()` that checks all entries of `contracts.json`.  This PR separates `_verify_single_precompiled_checksum()` that checks a single entry of `contracts.json`.

### Why I'm making this PR

After this PR, I want to expose a feature to check a single `contracts.json` entry.  That will be useful for [checking whether or not SecretRegistry's sources have changed](https://github.com/raiden-network/raiden-contracts/issues/1208#issuecomment-537490029).

### What's tricky about this PR (if any)

I had to take a moment to name the arguments of the new function.

----

Any reviewer can check these:

* [ ] In Python, use keyword arguments
* [ ] Squash unnecessary commits
* [ ] Comment commits
* [ ] Follow naming conventions
    * `python_variable`

And before "merge" all checkboxes have to be checked.  If you find redundant points, remove them.

### CODE REVIEW

File: raiden_contracts/contract_source_manager.py [hunk 1]
```diff
@@ -184,3 +182,25 @@ def check_runtime_codesize(d: Dict) -> None:
...
+    except KeyError:
+        raise ContractSourceManagerVerificationError(f"No checksum for {contract_name}")
+    if precompiled_checksum != expected_checksum:
+        raise ContractSourceManagerVerificationError(
+            f"checksum of {contract_name} does not match. got {precompiled_checksum} != "
+            "expected {expected_checksum}"
+        )
+
+
+def verify_single_precompiled_checksum_on_nonexistetnt_contract_name() -> None:
```
Comment (konradkonrad) [inline raiden_contracts/contract_source_manager.py]:
```suggestion
def verify_single_precompiled_checksum_on_nonexistent_contract_name() -> None:
```

File: raiden_contracts/tests/test_contracts_compilation.py [hunk 2]
```diff
@@ -256,3 +257,8 @@ def test_contract_source_manager_constructor_with_wrong_type() -> None:
     """ ConstructSourceManager's constructor raises TypeError on a wrong kind of argument """
     with pytest.raises(TypeError):
         ContractSourceManager(None)  # type: ignore
+
+
+def test_verify_single_precompiled_cyhecksum_on_nonexistent_contract_name() -> None:
+    with pytest.raises(ContractSourceManagerVerificationError, match="No checksum for"):
+        verify_single_precompiled_checksum_on_nonexistetnt_contract_name()
```
Comment (konradkonrad) [inline raiden_contracts/tests/test_contracts_compilation.py]:
```suggestion
        verify_single_precompiled_checksum_on_nonexistent_contract_name()
```

File: raiden_contracts/contract_source_manager.py (hunk 1 above)
Comment (konradkonrad) [inline raiden_contracts/contract_source_manager.py]: It appears, this functions exists only for testing behavior. So it seems a little odd to have it inside `contract_source_manager.py` -- can't you move it to, e.g. https://github.com/raiden-network/raiden-contracts/blob/master/raiden_contracts/tests/utils/contracts.py ?

File: raiden_contracts/tests/test_contracts_compilation.py [hunk 3]
```diff
@@ -22,6 +22,7 @@
     ContractSourceManager,
     ContractSourceManagerVerificationError,
     contracts_source_path,
+    verify_single_precompiled_checksum_on_nonexistetnt_contract_name,
```
Comment (konradkonrad) [inline raiden_contracts/tests/test_contracts_compilation.py]:
```suggestion
    verify_single_precompiled_checksum_on_nonexistent_contract_name,
```

File: raiden_contracts/contract_source_manager.py (hunk 1 above)
Comment (pirapira) [reply raiden_contracts/contract_source_manager.py]: I'm trying to avoid calling `_xyz` from other modules.  @hackaugusto what do you suggest?

**Pull request merged**